oauthproxy.go

```diff
@@ -810,19 +810,30 @@ func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.R
 // addHeadersForProxying adds the appropriate headers the request / response for proxying
 func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Request, session *sessionsapi.SessionState) {
 	if p.PassBasicAuth {
+		req.Header.Del("Authorization")
+		req.Header.Del("X-Forwarded-Email")
+		req.Header.Del("X-Forwarded-User")
```

Comment:
Why do we need to delete the `X-Forwarded-User` header when it's always set (on line 818)? Basically what's the benefit?

Reply:
The intent was simply to make each section the same - start with removing any headers that it purports to set, then set (sometimes conditionally, and sometimes, like this one, always) various values.